app/main.go

```diff
@@ -96,9 +105,62 @@ func main() {
 		log.Fatalf("Invalid flags: %v", err)
 	}
 
-	http.HandleFunc("/", index)
-	http.HandleFunc("/sync", sync)
-	http.HandleFunc("/cleanup", cleanup)
+	config := tsbridge.NewConfig(&tsbridge.ConfigOptions{
+		Filename:                 *metricConfig,
+		MinPointAge:              *minPointAge,
+		CounterResetInterval:     *counterResetInterval,
+		SDLookBackInterval:       *sdLookBackInterval,
+		SDInternalMetricsProject: *sdInternalMetricsProject,
+		UpdateParallelism:        *updateParallelism,
+		EnableStatusPage:         *enableStatusPage,
+		StorageEngine:            *storageEngine,
+		SyncPeriod:               *syncPeriod,
+		SyncCleanupAfter:         *syncCleanupAfter,
+	})
+
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		index(w, r, config)
+	})
+	http.HandleFunc("/sync", func(w http.ResponseWriter, r *http.Request) {
+		syncHandler(w, r, config)
+	})
+	http.HandleFunc("/cleanup", func(w http.ResponseWriter, r *http.Request) {
+		cleanupHandler(w, r, config)
+	})
+
+	// Run a sync loop for standalone use
+	if env.NotAppEngine() {
+		log.Debug("Running outside of appengine, starting up a sync loop...")
+		ctx, cancel := context.WithCancel(context.Background())
+		count := 0
+		go func() {
+			defer cancel()
+			for {
+				select {
+				case <-time.After(config.Options.SyncPeriod):
```

Comment:
Won't this make actual interval between sync operations longer than SyncPeriod (given that running sync takes non-zero amount of time)?

Reply:
Indeed, but I think it's an acceptable tradeoff between this and and spawning an extra routine every 10 minutes as that can cause race conditions on large amount of metrics (or issues + backoff that take >10m) unless we complicate this with locks.

WDYT?

Reply:
I think if users ask us to sync every 2 minutes, we should sync every 2 minutes; not 2+X minutes where X is outside of user's control).

There definitely should never be more than 1 syncing process running (since it could create out of order writes to Stackdriver), so I think the options here are:
1. measure how long syncing took, and then dynamically calculate how long to sleep for until the next sync (i.e. `sleep(sync_period - latency)`)
2. kick off syncing regularly, but check whether there is an existing sync running (a lock might be the easiest way to implement this). If there is an ongoing sync, just do nothing until the next timer. This is similar to how App Engine cron works, I believe.

Reply:
Gotcha. I've played around a bit with a couple of variants and couldn't find a simple one. Since async logic and timing are very constly in terms of maintenance and debugging, I've decided to add a TODO in here which I plan to resolve in another PR with a whole bunch of testing around it.